File: code_combat/7_Sarven_Desert/279-Minesweeper/minesweeper.js
// Lead the peasants and healer through the minefield.

while(true) {
    var coin = hero.findNearestItem();
    var healingThreshold = hero.maxHealth / 2;
    // Check to see if you are critically injured.
    if(hero.health < healingThreshold) {
        // Move left 10m.
        hero.moveXY(hero.pos.x - 10, hero.pos.y);
        // Ask for a heal.
        hero.say("Can I get a heal?");
    // Else, move to the next coin.
    } else if (coin) {
        hero.moveXY(coin.pos.x, coin.pos.y);
    }
}

File: projects/code_combat/7_Sarven_Desert/279-Minesweeper/minesweeper.py
while True:
    coin = hero.findNearestItem()
    if hero.health < hero.maxHealth / 2:
        hero.moveXY(hero.pos.x - 10, hero.pos.y)
        hero.say("Can I get a heal?")
    elif coin:
        hero.moveXY(coin.pos.x, coin.pos.y)
